### PR TITLE
fix pod created failed when resource request is 0

### LINKED ISSUE #109191: pod created failed when devices resource request is 0
### What happened?

1 create deployment by yaml below
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: jktest
spec:
  selector:
    matchLabels:
      workload.es.io: jktest
  template:
    metadata:
      labels:
        workload: jktest
    spec:
      containers:
      - image: centos:latest
        imagePullPolicy: IfNotPresent
        name: test
        resources:
          limits:
            cpu: 250m
            memory: 512Mi
            nvidia.com/gpu: "0"
          requests:
            cpu: 250m
            memory: 512Mi
            nvidia.com/gpu: "0"
```

2 create pod failed, the reason is 

```
status:
  message: >-
    Pod Node didn't have enough resource: nvidia.com/gpu, requestd: 0, used: 1,
    capacity: 0
  phase: Failed
  reason: OutOfnvidia.com/gpu
  startTime: '2022-02-02T12:40:19Z'  
```

### What did you expect to happen?

expect the pod create success, the request resource is 0, and it must not exceed

so it failed by somewhere unreasonable judgment

### How can we reproduce it (as minimally and precisely as possible)?

1 create pod `a` which use hot pluggable device in resource.request, and the resource is 1
2 remove device
3 create pod `b` which use hot pluggable device in resource.request, and the ressource is 0

### Anything else we need to know?

_No response_

### Kubernetes version

<details>

```console
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"20+", GitVersion:"v1.20.15", GitCommit:"4a89df5617b8e1e26abb16150502d04e6c180533", GitTreeState:"clean", BuildDate:"2022-03-26T06:34:08Z", GoVersion:"go1.15.15", Compiler:"gc", Platform:"linux/amd64"}
```

</details>

### Cloud provider

<details>

kubespray

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### PR DESCRIPTION
**Pull request by yylt**
fix pod created failed when resource request is 0

Signed-off-by: yylt <yang8518296@163.com>

#### Which issue(s) this PR fixes:

Fixes #109191

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/framework/plugins/noderesources/fit.go [hunk 1]
```diff
@@ -310,7 +310,8 @@ func fitsRequest(podRequest *preFilterState, nodeInfo *framework.NodeInfo, ignor
 				continue
 			}
 		}
-		if rQuant > (nodeInfo.Allocatable.ScalarResources[rName] - nodeInfo.Requested.ScalarResources[rName]) {
+		available := nodeInfo.Allocatable.ScalarResources[rName] - nodeInfo.Requested.ScalarResources[rName]
+		if rQuant > available && rQuant != 0 {
```
Comment (denkensk) [inline pkg/scheduler/framework/plugins/noderesources/fit.go:314]: `available` cannot be a negative number. So if `rQuant` > `available`, `rQuant`  must be larger than `0`.
So `&& rQuant != 0` is useless.

**Review by denkensk — COMMENTED** [review]
I think the original logic is right.

**Comment by jlsong01** [comment]
/close

sorry for my incorrect operation on the PR

**Pull request closed without merging**